File: April-18/27-04-18/java/01-Palindrome-string.java

import java.io.*;

class TestClass {
    static boolean pali(String str)
    {
        int n = str.length();
        for(int i = 0; i < n/2; i++)
        {
            if(str.charAt(i) != str.charAt(n-i-1))
                return false;
        }
        return true;
    }
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        String str = input.readLine();
        if(pali(str) == true)
            System.out.println("YES");
        else
            System.out.println("NO");
        
    }
}

File: April-18/27-04-18/python/01-Palindrome-string.py
def pali(s):
    n = len(s)
    for i in range(0, n//2):
        if(s[i] != s[n-i-1]):
            return False
    return True

data = input()
if(pali(data) == True):
    print("YES")
else:
    print("NO")
        
